river/tree/_base_tree.py

```diff
@@ -311,3 +322,244 @@ def __find_learning_nodes(self, node, parent, parent_branch, found):
+                'fontsize': '11', 'margin': '0.1,0.0'
+            },
+            edge_attr={'penwidth': '0.6', 'center': 'true', 'fontsize': '7  '}
+        )
+
+        if isinstance(self, base.Classifier):
+            n_colors = len(self.classes)   # noqa
+        else:
+            n_colors = 1
+
+        # Pick a color palette which maps classes to colors
+        new_color = functools.partial(next, iter(_color_brew(n_colors)))
+        palette = collections.defaultdict(new_color)
+
+        for parent_no, child_no, parent, child, branch_id in self._tree_root.iter_edges():
+
+            if child.depth > max_depth:
+                continue
+
+            if not child.is_leaf():
+                text = f'{child.split_test.attrs_test_depends_on()[0]}'
+
+                if child.depth == max_depth:
+                    text = f'{text}\n{node_prediction(child)}'
+            else:
+                text = f'{node_prediction(child)}\nsamples: {int(child.total_weight)}'
+
+            # Pick a color, the hue depends on the class and the transparency on the distribution
+            if isinstance(self, base.Classifier):
+                class_proba = {c: 0 for c in self.classes}   # noqa
+                sum_proba = sum(child.stats.values())
+                if sum_proba > 0:
+                    class_proba.update(
+                        normalize_values_in_dict(child.stats, factor=sum_proba, inplace=False)
+                    )
+                mode = max(class_proba, key=class_proba.get)
+                p_mode = class_proba[mode]
+                try:
+                    alpha = (p_mode - 1 / n_colors) / (1 - 1 / n_colors)
+                    fillcolor = str(transparency_hex(color=palette[mode], alpha=alpha))
+                except ZeroDivisionError:
+                    fillcolor = '#FFFFFF'
+            else:
+                fillcolor = '#FFFFFF'
+
+            dot.node(f'{child_no}', text, fillcolor=fillcolor, style='filled')
+
+            if parent_no is not None:
+                dot.edge(
+                    f'{parent_no}', f'{child_no}',
+                    xlabel=parent.split_test.describe_condition_for_branch(
+                        branch_id, shorten=True
+                    )
+                )
+
+        return dot
+
+
+# Utility adapted from the original creme's implementation
+def _color_brew(n: int) -> typing.List[typing.Tuple[int, int, int]]:
+    """Generate n colors with equally spaced hues.
+
+    Parameters
+    ----------
+    n
+        The number of required colors.
+
+    Returns
+    -------
+        List of n tuples of form (R, G, B) being the components of each color.
+    References
+    ----------
+    https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/tree/_export.py
+    """
+    colors = []
+
+    # Initialize saturation & value; calculate chroma & value shift
+    s, v = .75, .9
+    c = s * v
+    m = v - c
+
+    for h in [i for i in range(25, 385, int(360 / n))]:
```

Comment:
The `Hoeffding Adaptive Tree classifier` triggers an error:

```
~/Documents/workspace/git/river/river/tree/_base_tree.py in _color_brew(n)
    534     m = v - c
    535 
--> 536     for h in [i for i in range(25, 385, int(360 / n))]:
    537 
    538         # Calculate some intermediate values

ZeroDivisionError: division by zero
```

MWE:
```python
from river import synth
from river import evaluate
from river import metrics
from river import tree
gen = synth.Agrawal(classification_function=0, seed=42)
# Take 1000 instances from the infinite data generator
dataset = iter(gen.take(1000))
model = tree.HoeffdingAdaptiveTreeClassifier(
    grace_period=100,
    split_confidence=1e-5,
    nominal_attributes=['elevel', 'car', 'zipcode']
)
metric = metrics.Accuracy()
evaluate.progressive_val_score(dataset, model, metric)

model.draw()
```

Reply:
What a fortunate MWE! It was a bug. HATC was not updating the set of seen classes. I fixed that in commit [d878086](https://github.com/creme-ml/creme/pull/376/commits/d878086d72b3d364c71d36075caeff11b49a602a).